File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/101-SymmetricTree/101-SymmetricTree.cpp
// Last updated: 8/4/2026, 7:22:25 PM
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSymmetric(TreeNode* root) {
        if(root == nullptr) return true;
        return symmetry(root->left , root->right);
    }
    bool symmetry(TreeNode* left , TreeNode* right){
        if(left == nullptr && right == nullptr){
            return true;
        }
        if(left == nullptr || right == nullptr){
            return false;
        }

        if(left->val != right->val) return false;
        bool l = symmetry(left->left , right -> right);
        bool r = symmetry(left->right ,  right->left);

        return l && r;
    }
};

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/101-SymmetricTree/101-SymmetricTree.java
// Last updated: 8/4/2026, 7:22:32 PM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if(root == null) return true;
        return ismirror(root.left , root.right);
    }
    public boolean ismirror(TreeNode left , TreeNode right){
        if(left != null && right != null){
            boolean a = ismirror(left.left , right.right);
            boolean b = ismirror(left.right , right.left);
            if(left.val == right.val && a && b){
                return true;
            }
            else return false;
        }
        else if(left == null && right == null){
            return true;
        }
        else{
            return false;
        }
    }
}